x-pack/plugins/observability_solution/entity_manager/server/lib/entities/types.ts

```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EntityDefinition } from '@kbn/entities-schema';
+
+export type EntityDefinitionWithState = EntityDefinition & {
```

Comment:
i think it would be neat to include the transform health in here too (in a later PR)